### PR TITLE
Remove costly 'appendIfNotExists' and 'remove' call from PubSub

### PR DESCRIPTION
**Pull request by superkinglabs**
When there are thousands of subscriptions and the rate of subscribe and unsubscribe is high (hundreds per seconds), appendIfNotExists and remove call perform very poorly and was consuming all the CPU. Replacing the slice with the map improved the performance (CPU before was at 90% afterwards came down to 10%). I also had the pprof profile before and after this change but I am not able to find it now.

### CODE REVIEW

**Comment by vmihailenco** [comment]
Why would you subscribe/unsubscribe hundreds times per second?

**Comment by superkinglabs** [comment]
To build a presence system, whenever a user comes online, he needs to subscribe to all of his friends to receive their updates and similarly when he goes offline he will need to unsubscribe all of those.

**Comment by superkinglabs** [comment]
Also even if the rate of subscribe/unsubscribe is not very high, the map approach allows subscribe/unsubscribe to be O(1) without adding any extra space requirement. With the slice it would take linear time with each subscribe/unsubscribe.

File: pubsub.go [hunk 1]
```diff
@@ -67,12 +67,24 @@ func (c *PubSub) _conn(channels []string) (*pool.Conn, error) {
 func (c *PubSub) resubscribe(cn *pool.Conn) error {
 	var firstErr error
 	if len(c.channels) > 0 {
-		if err := c._subscribe(cn, "subscribe", c.channels...); err != nil && firstErr == nil {
+		channels := make([]string, len(c.channels), 0)
```
Comment (vmihailenco) [inline pubsub.go]: You need to remove `, 0` here and below to fix the build

Comment (superkinglabs) [reply pubsub.go]: Done. Also once you merge this, can you create a new release, I need this fix to use this package in my new project.

Comment (vmihailenco) [reply pubsub.go]: Released v6.10.2

**Pull request merged**